content/courses/program-security/duplicate-mutable-accounts.md

```diff
@@ -13,49 +13,38 @@ description:
   keys of the two accounts and throw an error if they are the same.
 
-  ```rust
-  if ctx.accounts.account_one.key() == ctx.accounts.account_two.key() {
-      return Err(ProgramError::InvalidArgument)
-  }
-  ```
+### Checking for Duplicate Mutable Accounts in Rust
 
-- In Anchor, you can use `constraint` to add an explicit constraint to an
-  account checking that it is not the same as another account.
+In Rust, you can simply compare the public keys of the accounts and return an
+error if they are identical:
+
+```rust
+if ctx.accounts.account_one.key() == ctx.accounts.account_two.key() {
+  return Err(ProgramError::InvalidArgument)
+}
+```
+
+### Using Constraints in Anchor
+
+In Anchor, you can add an explicit `constraint` to an account, ensuring it is
+not the same as another account.
 
 ## Lesson
 
-Duplicate Mutable Accounts refers to an instruction that requires two mutable
-accounts of the same type. When this occurs, you should validate that two
-accounts are different to prevent the same account from being passed into the
-instruction twice.
-
-Since the program treats each account as separate, passing in the same account
-twice could result in the second account being mutated in unintended ways. This
-could result in very minor issues, or catastrophic ones - it really depends on
-what data the code changes and how these accounts are used. Regardless, this is
-a vulnerability all developers should be aware of.
-
-#### No check
-
-For example, imagine a program that updates a `data` field for `user_a` and
-`user_b` in a single instruction. The value that the instruction sets for
-`user_a` is different from `user_b`. Without verifying that `user_a` and
-`user_b` are different, the program would update the `data` field on the
-`user_a` account, then update the `data` field a second time with a different
-value under the assumption that `user_b` is a separate account.
-
-You can see this example in the code below.Tthere is no check to verify that
-`user_a` and `user_b` are not the same account. Passing in the same account for
-`user_a` and `user_b` will result in the `data` field for the account being set
-to `b` even though the intent is to set both values `a` and `b` on separate
-accounts. Depending on what `data` represents, this could be a minor unintended
-side-effect, or it could mean a severe security risk. allowing `user_a` and
-`user_b` to be the same account could result in
+**Duplicate Mutable Accounts** occur when an instruction requires two mutable
+accounts of the same type. If the same account is passed twice, it can be
+mutated in unintended ways, potentially causing security vulnerabilities.
+
+### No check
+
+Consider a program that updates a data field for `user_a` and `user_b` in a
+single instruction. If the same account is passed for both `user_a` and
+`user_b`, the program will overwrite the data field with the second value,
+potentially leading to unintended side effects.
```

Comment:
Excellent @0xCipherCoder - this has a lot higher signal:noise ratio than the older version.